core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/CasAuthenticationEventListenerTests.java

```diff
@@ -112,43 +129,137 @@ public void verifyCasAuthenticationTransactionFailureEvent() {
+        assertRepositoryIsEmpty();
         val event = new CasTicketGrantingTicketDestroyedEvent(this,
-            new MockTicketGrantingTicket("casuser"));
-        applicationContext.publishEvent(event);
+                new MockTicketGrantingTicket("casuser"), ClientInfoHolder.getClientInfo());
+        publishEventAndWaitToProcess(event);
+        assertFalse(casEventRepository.load().findAny().isEmpty());
+    }
+
+
+    @Test
+    public void verifyEventRepositoryHasOneEventOnly() {
+        assertRepositoryIsEmpty();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"), ClientInfoHolder.getClientInfo());
+        publishEventAndWaitToProcess(event);
+        assertEquals(1, casEventRepository.load().count());
+    }
+
+    @Test
+    public void verifyCasTicketGrantingTicketDestroyedHasClientInfo() {
+        assertRepositoryIsEmpty();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"), ClientInfoHolder.getClientInfo());
+        publishEventAndWaitToProcess(event);
         assertFalse(casEventRepository.load().findAny().isEmpty());
+        val result = casEventRepository.load().toList().get(0).getClientIpAddress();
+        assertEquals(REMOTE_ADDR_IP, result);
+    }
+
+    @Test
+    public void verifyCasTicketGrantingTicketDestroyedHasClientInfoWithMultipleThreads() throws Exception {
+        assertRepositoryIsEmpty();
+        var currentCount = casEventRepository.load().count();
+        val threadPool = Executors.newFixedThreadPool(THREAD_POOL_SIZE);
+        val futureList = new ArrayList<Future<Integer>>();
+        var expectedNumOfIp1 = 0;
+        for (var x = 0; x < NUM_OF_REQUESTS; x++) {
+            if (shouldUseIp1(x)) {
+                expectedNumOfIp1++;
+            }
+            futureList.add(threadPool.submit(new HttpServletRequestSimulation(x, shouldUseIp1(x), applicationContext)));
+        }
+        var maxThread = -1;
+        for (var future : futureList) {
+            var currentThread = future.get();
+            if (currentThread > maxThread) {
+                maxThread = currentThread;
+            }
+        }
+
+        waitForSpringEventToProcess(currentCount + maxThread + 1);
+        val eventSize = (int) casEventRepository.load().count();
+        val numOfIp1s = (int) casEventRepository.load().filter(e -> HttpServletRequestSimulation.IP1.equals(e.getClientIpAddress())).count();
+        assertEquals(maxThread + 1, eventSize);
+        assertEquals(expectedNumOfIp1, numOfIp1s);
+    }
+
+    private boolean shouldUseIp1(final int x) {
+        return x % NUM_TO_USE_IP1 == 0;
+    }
+
+    private void clearEventRepository() {
+        casEventRepository.removeAll();
+    }
+
+    /**
+     * Pass in the number of expected events that should have been stored in the CasEventRepository after publishing a new event.
```

Comment:
For all javadocs, references to interfaces, classes and components need to be formatted as links using `@link`. Otherwise, when things change or are renamed, we won't catch them automatically.

Reply:
thanks for the tip.. It should be updated now.